yak/src/mc/marching_cubes.cpp

```diff
@@ -150,55 +157,61 @@ std::vector<Triangle> processCube(const yak::TSDFContainer& grid, int x, int y,
 
 pcl::PolygonMesh makeMesh(const yak::TSDFContainer& grid, const yak::MarchingCubesParameters& params)
 {
-  const int max_threads = omp_get_max_threads();
-  std::vector<std::vector<Triangle>> triangle_buffers(max_threads);
+  vtkSmartPointer<vtkPoints> points = vtkSmartPointer<vtkPoints>::New();
+  vtkSmartPointer<vtkCellArray> triangles = vtkSmartPointer<vtkCellArray>::New();
 
 #pragma omp parallel for
   for (int x = 1; x < grid.dims().x(); ++x)
   {
-    const int tid = omp_get_thread_num();
-
     for (int y = 1; y < grid.dims().y(); ++y)
     {
       for (int z = 1; z < grid.dims().z(); ++z)
       {
         std::vector<Triangle> ts = processCube(grid, x - 1, y - 1, z - 1, params.min_weight);
-        triangle_buffers[tid].insert(triangle_buffers[tid].end(), ts.begin(), ts.end());
+#pragma omp critical
+        {
+          for (auto& t : ts)
+          {
+            vtkIdType p1 = points->InsertNextPoint(t.v[0].x(), t.v[0].y(), t.v[0].z());
+            vtkIdType p2 = points->InsertNextPoint(t.v[1].x(), t.v[1].y(), t.v[1].z());
+            vtkIdType p3 = points->InsertNextPoint(t.v[2].x(), t.v[2].y(), t.v[2].z());
```

Comment:
These lines produce a warning about implicit conversion from float to double. Can you add some static casts around `t.v[0].x()`, etc.?

Reply:
Changes have been made.